### PR TITLE
fix simplex in mosek

### PR DESCRIPTION
**Pull request by goriri**
WARNING: code from python novice. Trying to fix format_results exception when using mosek with simplex method.

### CODE REVIEW

File: cvxpy/problems/solvers/mosek_intf.py [hunk 1]
```diff
@@ -273,8 +274,20 @@ def format_results(self, task, data, cached_data):
                       mosek.solsta.near_dual_infeas_cer: s.UNBOUNDED_INACCURATE,
                       mosek.solsta.unknown: s.SOLVER_ERROR}
 
-        task.getprosta(mosek.soltype.itr)  # unused
-        solsta = task.getsolsta(mosek.soltype.itr)
+        soltype = None
+        for sol in [mosek.soltype.itr, mosek.soltype.bas]:
```
Comment (SteveDiamond) [inline cvxpy/problems/solvers/mosek_intf.py]: What is this doing? Can you add comments?

**Comment by SteveDiamond** [comment]
The code looks reasonable but I need more explanatory comments to be sure. Travis CI failed because of style errors. Run flake8 on your code and fix mosek_intf.py.

I'm having contributors sign a CLA now via this license/cla thing. It should be easy. Let me know if you have any problems.

File: cvxpy/problems/solvers/mosek_intf.py (hunk 1 above)
Comment (goriri) [inline cvxpy/problems/solvers/mosek_intf.py]: For simplex solution, the mosek.soltype.itr is empty but the actual solution lies in mosek.soltype.bas. I cannot find a way to deduce the soltype here so trying to fall back to mosek.soltype.bas if mosek.soltype.itr is not working.

Comment (RhysU) [inline cvxpy/problems/solvers/mosek_intf.py]: This is not unlike https://github.com/cvxgrp/cvxpy/issues/335.

**Comment by SteveDiamond** [comment]
Sorry for not responding for so long. #335 seems quite related, and I prefer the suggestion there to use  "task.solutiondef" rather than a try catch block. Could you update the pull request along those lines?

**Comment by SteveDiamond** [comment]
This was resolved by PR #347

**Pull request closed without merging**